scripts/run_frontend_tests.py

```diff
@@ -108,6 +119,35 @@ def run_dtslint_type_tests() -> None:
         sys.exit('The dtslint (type tests) failed.')
 
 
+def get_file_spec(file_path: str) -> str | None:
+    """Returns the spec file path for a given file path.
+
+    Args:
+        file_path: str. The path of the file.
+
+    Returns:
+        str | None. The path of the spec file if it exists, otherwise None.
+    """
+    if (
+        file_path.endswith(('.spec.ts', '.spec.js', 'Spec.ts', 'Spec.js')) and
+        os.path.exists(file_path)
+    ):
+        return file_path
+
+    if file_path.endswith(('.ts', '.js')):
+        spec_file_path = file_path.replace('.ts', '.spec.ts').replace(
```

Comment:
I think something like this would be better:

```suggestion
        spec_file_path = '%s.spec%s' % (file_path[:-3], file_path[-3:])
```

This will avoid unexpected behavior from something like `oppia.tsconfig.js`

Reply:
Done, Thanks.